### DISCUSSION TITLE
WASM Build?

### DISCUSSION THREAD

**Discussion by tmcw** [oxc-project/oxc#3311] [Feature Request]
Hi! I noticed that the [oxc project playground](https://oxc-project.github.io/oxc/playground/) is using a WASM build of oxc, but I can't find any packages, discussion, or releases of an official build. For folks building more client-side tooling, it'd be super nifty to be able to run oxc in a browser like that.

**Comment by Boshen** [comment]
The parser has a wasm version published to https://www.npmjs.com/package/@oxc-parser/wasm

Everything else is still unstable so they are not published.

**Comment by Boshen** [comment]
The npm packages `oxc-parser`, `oxc-transform` and `oxc-minify` all have a wasm build. See https://stackblitz.com/edit/oxc-parser for usage example.

**Comment by sandstrom** [comment]
@Boshen Are there plans to add WASM builds for `oxc-resolver`, `oxlint` or `oxfmt`? 

Webassembly builds make these tools easily accessible for embedding in browsers and other fun things with web assembly.

They all seem stable, for example oxfmt recently entering beta.

For example, esbuild has had support for a long time:

https://github.com/evanw/esbuild/blob/main/Makefile#L346-L347

If you add it, I would target WASI 0.2 (preview 2, or `wasip2`), see https://wasi.dev/interfaces#wasi-02.
